geoserver/geoserver_seeder/layer_seeder.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/ruby
+
+require 'nokogiri'
+require 'open-uri'
+require 'trollop'
+require 'net/http'
+require 'net/https'
+require 'open-uri'
+require 'logger'
+
+
+class LayerSeeder
+    def initialize(layer, opts)
+        @geowebcache = Geowebcache.new()
+        @layer = layer
+        @start_zoom = opts[:start_zoom]
+        @end_zoom = opts[:end_zoom]
+        @url_format = opts[:url_format]
+        @version = opts[:version] || "1.1.1"
+        @tile_size = opts[:tile_size]
+        @gutter_size = opts[:gutter_size]
+        @geoserver_wms_url = opts[:geoserver_wms_url]
+        @dry_run = opts[:dry_run]
+        @purge = opts[:purge]
+    end
+
+
+    def purgeCache
+
+        geowebcacheUrl = "#{@geowebcache.url}/rest/seed/#{@layer[1]}"
+
+        options = "<seedRequest>"\
+            "<name>#{@layer[1]}</name>"\
+              "<srs>"\
+                "<number>4326</number>"\
+                    "</srs>"\
+              "<zoomStart>1</zoomStart>"\
+              "<zoomStop>20</zoomStop>"\
+              "<format>image/png</format>"\
+              "<type>truncate</type>"\
+           "</seedRequest>"
+
+        # kill all previous threads for layer
+        killcmd = "curl -v -u #{@geowebcache.username}:#{@geowebcache.password}  -d \"kill_all=all\"  \"#{geowebcacheUrl}\""
+        # clear all tiles for layer
+        cmd = "curl -v -u #{@geowebcache.username}:#{@geowebcache.password} -XPOST -H \"Content-type: text/xml\" -d \"#{options}\" \"#{geowebcacheUrl}.xml\""
+
+        if @dry_run
+            $logger.info("(Dry run kill) #{cmd}")
+            $logger.info("(Dry run truncate) #{killcmd}")
+        else
+            system(killcmd)
+            system(cmd)
+            $logger.info("(Excecuting kill->truncate) #{geowebcacheUrl}")
+        end
+    end
+
+
+    def seedCache
+
+        geowebcacheUrl = "#{@geowebcache.url}/rest/seed/#{@layer[1]}"
+
+        options = "<seedRequest>"\
+            "<name>#{@layer[1]}</name>"\
+              "<srs>"\
+                "<number>4326</number>"\
+                    "</srs>"\
+              "<zoomStart>#{@start_zoom}</zoomStart>"\
+              "<zoomStop>#{@end_zoom}</zoomStop>"\
+              "<format>image/png</format>"\
+              "<type>seed</type>"\
+              "<threadCount>2</threadCount>"\
+           "</seedRequest>"
+
+        cmd = "curl -v -u #{@geowebcache.username}:#{@geowebcache.password} -XPOST -H \"Content-type: text/xml\" -d \"#{options}\" \"#{geowebcacheUrl}.xml\""
+
+        if @dry_run
+            $logger.info("(Dry run seed) #{cmd}")
+        else
+            system(cmd)
+            $logger.info("(Excecuting seed) #{geowebcacheUrl}")
+        end
+    end
+
+    def execute
+
+
+        if @type == "truncate"
+            purgeCache
+        else
+            purgeCache
```

Comment:
I would say for the most part the seeder will be called programatically, the same for each layer. We would also have our eye on how long it takes to seed a layer on average and would keep this small so that we can handle hundreds of layers in a concise amount of time. For this reason i dont think we need to preserve cache to add higher levels